src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -404,9 +411,36 @@ internal T WithProofOfPosession(HttpRequestMessage httpRequestMessage, IPoPCrypt
             }
 
             CommonParameters.AddApiTelemetryFeature(ApiTelemetryFeature.WithPoPScheme);
-            CommonParameters.AuthenticationScheme = new PoPAuthenticationScheme(httpRequestMessage, popCryptoProvider);
+            CommonParameters.AuthenticationScheme = new PoPAuthenticationScheme(httpRequestMessage.RequestUri, 
+                                                                                httpRequestMessage.Method, 
+                                                                                popCryptoProvider);
+
+            return (T)this;
+        }
+
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <param name="popAuthenticationConfiguration"> </param>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item> An Authentication header is automatically added to the request</item>
+        /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
+        /// <item>MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</item>
+        /// </list>
+        /// </remarks>
+        public T WithProofOfPosession(PopAuthenticationConfiguration popAuthenticationConfiguration)
```

Comment:
IMO this should be the only POP method. Let's delete the old one.

Reply:
that's a breaking change?
would we do that because it's an experimental feature?

Reply:
Yes, experimental features can introduce breaking changes at any time.